packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts

```
@@ -612,6 +675,66 @@ function assertNoComplexSizes(dir: NgOptimizedImage) {
   }
 }
 
+/**
+ * Verifies that placeholderConfig isn't being used without placeholder
+ */
+function assertNoPlaceholderConfigWithoutPlaceholder(dir: NgOptimizedImage) {
+  if (dir.placeholderConfig && !dir.placeholderConfig.blur && !dir.placeholder) {
+    throw new RuntimeError(
+        RuntimeErrorCode.INVALID_INPUT,
+        `${
+            imgDirectiveDetails(
+                dir.ngSrc, false)} \`placeholderConfig\` options were provided for an ` +
+            `image that does not use the \`placeholder\` attribute, and will have no effect.`);
+  }
+}
+
+/**
+ * Warns if a relative URL placeholder is specified, but no loader is present to provide the small
+ * image.
+ */
+function assertNoRelativePlaceholderWithoutLoader(dir: NgOptimizedImage, imageLoader: ImageLoader) {
```

Q: Should we throw here instead of warning? We might want to consider it since no placeholder is being served.

---

Originally I was doing a `console.warn` to make it consistent with the other "this doesn't do anything without a loader" assertion for `srcset`. But I think you're right that throwing is better, so I went ahead and made this throw. Too late to fix the srcset one, though, since that would be a breaking change.